bundles/core/org.eclipse.smarthome.core.library/src/main/java/org/eclipse/smarthome/core/library/types/GenericComplexType.java

```diff
@@ -0,0 +1,90 @@
+/**
+ * Copyright (c) 2014-2015 openHAB UG (haftungsbeschraenkt) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+
+package org.eclipse.smarthome.core.library.types;
+
+import java.util.Formatter;
+import java.util.SortedMap;
+import java.util.TreeMap;
+
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.ComplexType;
+import org.eclipse.smarthome.core.types.PrimitiveType;
+import org.eclipse.smarthome.core.types.State;
+
+
+/**
+ * This type can be used for items that are dealing with telephony functionality.
+ * 
+ * @author Thomas.Eichstaedt-Engelen / Initially CallType
+ * @author Gaël L'hopital - port to Eclipse SmartHome
+ * 
+ */
+public class GenericComplexType implements ComplexType, Command, State {
+	
+	private SortedMap<String, PrimitiveType> typeDetails = new TreeMap<String, PrimitiveType>();
+	
+	public GenericComplexType(StringType...elements) {
+		for (StringType element : elements) {
+			String[] parts = element.toString().split("=");
+			if (elements.length == 2) {
```

Comment:
Ah, I forgot that not only one key value pair must be (de)serialized, but also the all entries. To accomplish this, we have also to limit the value a little bit or be a little bit clever.
So, we could separate a key-value pair using the schema above.
The first '=' will split the key-value-string, between key (character sequence in front of first '=') and value (remaining part).
One key value pair could be separated by another key-value pair using ';'.
We could limit the key, to not contain '=' and ';'.
To keep the semicolon available as valid character in the value part, we could escape all semicolon in the value (replace ';' by '\;') on serialization only.
For deserialization, we split on semicolons that are not escaped and then split on first '=' ...

<pre>
key1=value1;key2=value2;key3=value=with=foo;key4=value\;with\;foo;key5=;key6=\;\;foo
</pre>

I think the example above is human readable...